src/main/kotlin/com/team4099/robot2023/subsystems/groundintake/GroundIntake.kt

```diff
@@ -115,10 +116,10 @@ class GroundIntake(private val io: GroundIntakeIO) : SubsystemBase() {
 
   var lastIntakeRunTime = Clock.fpgaTime
 
-  var currentState: GroundIntakeConstants.ArmStates = GroundIntakeConstants.ArmStates.UNINITIALIZED
+  var currentState: GroundIntakeState = GroundIntakeState.UNINITIALIZED
 
-  var requestedState: GroundIntakeConstants.ArmStates =
-    GroundIntakeConstants.ArmStates.TARGETING_POSITION
+  var requestedState: GroundIntakeRequest =
```

Comment:
can we name this `currentRequest` or something instead? the point I have been trying to hammer home is you aren't really requesting a state, you're requesting an action. In this case the actions are just states but for clarity you should rename